libfastpm/cosmology.c

```diff
@@ -75,17 +50,17 @@ double Omega_r(FastPMCosmology * c)
     return Omega_g(c) + Omega_ur(c);
 }
 
-double getFtable(int F_id, double y)
+double getFtable(int F_id, double y, FastPMCosmology * c)
 {
     //Not using size as an arg, it's globally defined
     //Gets the interpolated value of Ftable[F_id] at y
     //F_id: 1 for F, 2 for F', 3 for F''
     
-    if (y == 0.) {               //does this work for double?
-        return 0.;                //this hack ensures all Omega_ncdm related funcs will return 0.
-    }else{
-    return interpolate(Ftable[0], Ftable[F_id], Fsize, y);
-    }
+    //if (y == 0.) {               //does this work for double?
+    //    return 0.;                //this hack ensures all Omega_ncdm related funcs will return 0.
+    //}else{
+    return fastpm_do_fd_interp(c->FDinterp, F_id, y);
+    //}
 }
 
 double Fconst(int ncdm_id, FastPMCosmology * c)
```

Comment:
Would consume FastPMFDInterp -- actually, what about calling the class 'FastPMFermiDirac'? Interp or not is just an internal detail.